src/components/Account/AccountDetails.vue

```diff
@@ -0,0 +1,177 @@
+<template>
+  <div class="account-details">
+    <form v-if="user" @submit.prevent="saveChanges">
+      <CustomInput
+        ref="emailInput"
+        class="account-details__input"
+        v-model="user.email.inputValue"
+        label="Email"
+        placeholder="Something ending with monterail.com"
+        type="email"
+        :rules="emailRules"
+        @updateValidation="(value) => (user.email.isValid = value)"
```

Comment:
[just a note] it seems to be a good case for the second `v-model`, wdyt? 🤔  

```vue
v-model="user.email.inputValue"
v-model:validation="user.email.isValid"
```

another way to make it a bit shorter is to write
```vue
@updateValidation="user.email.isValid = $event"
```

Reply:
I didn't know I can set two or more v-models on one element. That's a game changer for me! Thanks!
I will try to implement it in a separate PR.